lib/modules/dosomething/dosomething_taxonomy/dosomething_taxonomy.module

```diff
@@ -110,15 +110,21 @@ function dosomething_taxonomy_preprocess_taxonomy_term(&$vars) {
 
 /**
  * Returns array of node nid's with the given Term $tid.
- * @deprecated Use taxonomy_select_nodes instead.
  *
  * @param int $tid
  *   A taxonomy term tid.
  *
  * @return array
  */
 function dosomething_taxonomy_get_campaigns($tid) {
-  return taxonomy_select_nodes($tid, FALSE);
+  $nodes = node_load_multiple(taxonomy_select_nodes($tid, FALSE));
```

Comment:
if you're looking for a specific type of node (it seems we are here) you can add a `type` to `node_load_multiple` to only pull campaign nodes